test/integration/scoped_rds_integration_test.cc

```diff
@@ -470,6 +470,34 @@ route_configuration_name: foo_route1
   cleanupUpstreamAndDownstream();
 }
 
+TEST_P(ScopedRdsIntegrationTest, RejectKeyConflictInDeltaUpdate) {
+  if (!isDelta()) {
+    return;
+  }
+  const std::string scope_route1 = R"EOF(
+name: foo_scope1
+route_configuration_name: foo_route1
+key:
+  fragments:
+    - string_key: foo
+)EOF";
+  on_server_init_function_ = [this, &scope_route1]() {
+    createScopedRdsStream();
+    sendSrdsResponse({scope_route1}, {scope_route1}, {}, "1");
+  };
+  initialize();
+  // Delta SRDS update with key conflict, should be rejected.
+  const std::string scope_route2 = R"EOF(
+name: foo_scope2
+route_configuration_name: foo_route1
+key:
+  fragments:
+    - string_key: foo
+)EOF";
+  sendSrdsResponse({scope_route1, scope_route2}, {scope_route2}, {}, "2");
```

Comment:
do you want to check on some stats to see if these responses fail or not?

Reply:
+1.
That makes sense, otherwise how will the test observe a failure (i.e., that the conflict wasn't rejected).

Reply:
Makes sense.